### PR TITLE
[cli] Preparing release 4.1.0

### PR DESCRIPTION
**Pull request by wrwg**
## Description

Prepare for CLI release 4.1. This makes compiler v2 stable and adds a few new tools.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [x] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Manually tested that `--move-2` is not producing a warning about unstable compiler or language. 

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by wrwg** [comment]
* **#14462** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/14462?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wrwg and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rahxephon89 — APPROVED** [review]

**Review by brmataptos — APPROVED** [review]

File: crates/aptos/CHANGELOG.md [hunk 1]
```diff
@@ -3,9 +3,15 @@
 All notable changes to the Aptos CLI will be captured in this file. This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html) and the format set out by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-- Add safe methods to delete a profile, to rename a profile, and to output the private key of a profile.
-- Add `aptos update movefmt`. This installs / updates the `movefmt` binary, which is needed for the new `aptos move fmt` subcommand.
-- Integrate the Move formatter `movefmt` which is now available via `aptos move fmt`
+
+## [4.1.0] - 2024/08/30
+- Marks Move 2 and new compiler as stable.
```
Comment (vineethk) [inline crates/aptos/CHANGELOG.md:8]:
```suggestion
- Marks Move 2 and compiler v2 as stable.
```

File: crates/aptos/CHANGELOG.md [hunk 2]
```diff
@@ -3,9 +3,15 @@
 All notable changes to the Aptos CLI will be captured in this file. This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html) and the format set out by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
-- Add safe methods to delete a profile, to rename a profile, and to output the private key of a profile.
-- Add `aptos update movefmt`. This installs / updates the `movefmt` binary, which is needed for the new `aptos move fmt` subcommand.
-- Integrate the Move formatter `movefmt` which is now available via `aptos move fmt`
+
```
Comment (vineethk) [inline crates/aptos/CHANGELOG.md:6]: Will the updates to `crates/aptos/Cargo.toml` and `Cargo.lock` (for package "aptos") be coming in a different PR?

Usually, the version bumps and changelog are done in one PR, from what I have seen so far.

Comment (wrwg) [reply crates/aptos/CHANGELOG.md:6]: Good catch, done.
